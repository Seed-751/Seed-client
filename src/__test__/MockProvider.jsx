import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import GrobalStyles from "../styles/index";
import theme from "../styles/theme";
import store from "../reducers/store";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export default function MockProvider({ children }) {
  const history = createMemoryHistory();

  return (
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>
          <GrobalStyles />
          {children}
        </ThemeProvider>
      </Router>
    </Provider>
  );
}

MockProvider.propTypes = {
  children: PropTypes.element.isRequired,
};
