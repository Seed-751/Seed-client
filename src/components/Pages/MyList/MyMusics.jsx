import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import PropTypes from "prop-types";

import Music from "../../Shared/Music";

const Container = styled.div`
  padding: 20px;
`;

const MusicBox = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  row-gap: 20px;
  column-gap: 20px;
  margin-top: 10px;

  .link {
    width: 250px;
  }
`;

export default function MyMusics({ myMusics }) {
  return (
    <Container>
      <strong>My Musics</strong>
      <MusicBox>
        {myMusics?.map((music) => (
          <Link key={music._id} to={`/musics/${music._id}`} className="link">
            <Music key={music._id} music={music} />
          </Link>
        ))}
      </MusicBox>
    </Container>
  );
}

MyMusics.propTypes = {
  myMusics: PropTypes.array,
};
