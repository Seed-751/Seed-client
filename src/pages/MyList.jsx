import React, { useState, useEffect } from "react";
import { useDispatch } from "react-redux";

import requestMyMusics from "../api/requestMyMusics";
import requestMyFundings from "../api/requestMyFundings";
import { occurError } from "../reducers/noticeSlice";
import MyMusics from "../components/MyMusics";
import MyFundings from "../components/MyFundings";

export default function MyList() {
  const dispatch = useDispatch();
  const [myMusics, setMyMusics] = useState([]);
  const [myFundings, setMyFundings] = useState([]);

  useEffect(() => {
    async function getMyMusics() {
      try {
        const { data, message } = await requestMyMusics();

        if (data) {
          return setMyMusics(data);
        }

        dispatch(occurError(message));
      } catch (err) {
        dispatch(occurError(err.message));
      }
    }

    async function getMyFundings() {
      try {
        const { data, message } = await requestMyFundings();

        if (data) {
          return setMyFundings(data);
        }

        dispatch(occurError(message));
      } catch (err) {
        dispatch(occurError(err.message));
      }
    }

    getMyMusics();
    getMyFundings();
  }, [dispatch]);

  return (
    <>
      <MyMusics myMusics={myMusics} />
      <MyFundings myFundings={myFundings} />
    </>
  );
}
