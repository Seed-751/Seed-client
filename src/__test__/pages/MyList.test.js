import React from "react";
import { render } from "@testing-library/react";

import MyList from "../../pages/MyList";
import MockProvider from "../MockProvider";

describe("MyList page", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <MockProvider>
        <MyList />
      </MockProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
