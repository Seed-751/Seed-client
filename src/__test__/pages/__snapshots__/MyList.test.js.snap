// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList page matches snapshot 1`] = `
<div>
  <div
    class="sc-eCstlR cxNDuA"
  >
    <h1>
      My Musics
    </h1>
    <div
      class="sc-jSgvzq dSGgCD"
    />
  </div>
  <div
    class="sc-gKseQn efNTNe"
  >
    <h1>
      My Fundings
    </h1>
    <div
      class="sc-iBPTik bpAjIl"
    />
  </div>
</div>
`;
